lib/maths/CSeasonalComponent.cc

```diff
@@ -210,29 +207,18 @@ double CSeasonalComponent::delta(core_t::TTime time,
     core_t::TTime longPeriod{time_.period()};
 
     if (longPeriod > shortPeriod && longPeriod % shortPeriod == 0) {
-        TMinAccumulator min;
-        TMinMaxAccumulator minmax;
+        TMinMaxAccumulator bias;
+        TMinMaxAccumulator cancellation;
+        cancellation.add(-shortDifference);
         double mean{this->CDecompositionComponent::meanValue()};
         for (core_t::TTime t = time; t < time + longPeriod; t += shortPeriod) {
             if (time_.inWindow(t)) {
-                double difference{CBasicStatistics::mean(this->value(t, 0.0)) - mean};
-                min.add(std::fabs(difference));
-                minmax.add(difference);
+                double longDifference{CBasicStatistics::mean(this->value(t, 0.0)) - mean};
+                bias.add(longDifference);
+                cancellation.add(longDifference);
             }
         }
-
-        if (std::fabs(minmax.signMargin()) > 0.0) {
-            return minmax.signMargin();
-        }
-
-        // We have a smooth decision boundary for whether to apply
-        // a delta for the case that the difference from the mean
-        // is 1/3 of the range. We force the delta to zero for values
-        // significantly smaller than this.
-        double scale{CTools::logisticFunction(3.0 * min[0] / minmax.range(), 0.1, 1.0)};
-        scale = CTools::truncate(1.002 * scale - 0.001, 0.0, 1.0);
-
-        return -scale * min[0] * CTools::sign(shortPeriodValue);
+        return bias.signMargin() != 0.0 ? bias.signMargin() : cancellation.signMargin();
```

Comment:
Would you want to treat very small numbers as 0?  For example, if `bias.signMargin()` was 1e-308 would you still want to use it?  (This is only a question - if the answer is yes then fine.)

Reply:
One could, for example, not bother if it is << amplitude. The chances of this being very small, but not identically zero, is actually small. If there is no bias then we expect the differences at the different times in the period to *not* all have the same sign with a high probability because noise in the underlying values should cause them to randomly land on different sides of the mean. Also, it doesn't have a significant impact to have a small, but non-zero, delta. So on balance I don't think this warrants the extra complication.